src/app/components/filebrowsermvs/filebrowsermvs.component.ts

```diff
@@ -127,13 +132,146 @@ export class FileBrowserMVSComponent implements OnInit, OnDestroy {//IFileBrowse
+      } else {
+        this.deleteNonVsamDataset(rightClickedFile);
+      }
+    });
+  }
+
+  deleteNonVsamDataset(rightClickedFile: any): void {
+    this.isLoading = true;
+    this.deletionQueue.set(rightClickedFile.data.path, rightClickedFile);
+    rightClickedFile.styleClass = "filebrowsermvs-node-deleting";
+    let deleteSubscription = this.datasetService.deleteNonVsamDatasetOrMember(rightClickedFile)
+    .subscribe(
+      resp => {
+        this.isLoading = false;
+        this.snackBar.open(resp.msg,
+        'Dismiss', { duration: 5000,   panelClass: 'center' });
+        this.removeChild(rightClickedFile);
+        this.deletionQueue.delete(rightClickedFile.data.path);
+        rightClickedFile.styleClass = "";
+      },
+      error => {
+        if (error.status == '500') { //Internal Server Error
+          this.snackBar.open('Failed to delete: ' + rightClickedFile.data.path + "' This is probably due to a server agent problem.", 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+        } else if (error.status == '404') { //Not Found
+          this.snackBar.open(rightClickedFile.data.path + ' has already been deleted or does not exist.', 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+          this.removeChild(rightClickedFile);
+        } else if (error.status == '400') { //Bad Request
+          this.snackBar.open("Failed to delete '" + rightClickedFile.data.path + "' This is probably due to a permission problem.", 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+        } else { //Unknown
+          this.snackBar.open("Uknown error '" + error.status + "' occured for: " + rightClickedFile.data.path, 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+          // Error info gets printed in uss.crud.service.ts
+        }
+        this.deletionQueue.delete(rightClickedFile.data.path);
+        this.isLoading = false;
+        rightClickedFile.styleClass = "";
+        this.errorMessage = <any>error;
+      }
+    );
+
+    setTimeout(() => {
+      if (deleteSubscription.closed == false) {
+        this.snackBar.open('Deleting ' + rightClickedFile.data.path + '... Larger payloads may take longer. Please be patient.', 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+      }
+    }, 4000);
+  }
+
+  deleteVsamDataset(rightClickedFile: any): void {
+    this.isLoading = true;
+    this.deletionQueue.set(rightClickedFile.data.path, rightClickedFile);
+    rightClickedFile.styleClass = "filebrowsermvs-node-deleting";
+    let deleteSubscription = this.datasetService.deleteVsamDataset(rightClickedFile)
+    .subscribe(
+      resp => {
+        this.isLoading = false;
+        this.snackBar.open(resp.msg,
+        'Dismiss', { duration: 5000,   panelClass: 'center' });
+        //Update vs removing node since symbolicly linked data/index of vsam can be named anything
+        this.updateTreeView(this.path);
```

Comment:
Wouldn't this collapse existing expanded DS folders? It may be a necessary evil, but I'm wondering if there's a smarter way to do it than doing a fresh update, as deleting anything VSAM could really screw with people's workflows. We should at least create a ticket, for a workaround, if the fix becomes out of scope